frontend/src/app/core/helpers/time-zone-helper.ts

```diff
@@ -1,18 +1,63 @@
-import { formatDate } from '@angular/common';
 import { ITimeZone } from '@core/models/ITimeZone';
 import { ILocalUser } from '@core/models/IUser';
 
+export const millisInHour = 3600 * 1000;
+export const millisInMinute = 60 * 1000;
+
 export const getDefaultTimeZone = (): ITimeZone => {
     const user = JSON.parse(localStorage.getItem('user')!) as ILocalUser;
 
     return user?.timeZone ?? { nameValue: '', timeValue: '' };
 };
 
-export const getDateWithoutLocalOffset = (date: Date) =>
-    new Date(date.getTime() - date.getTimezoneOffset() * 60000);
+/**
+ * Returns number of milliseconds in time zone
+ *
+ * @example
+ * +01:00 -> 3600000
+ * -02:30 -> -9000000
+ */
+export const parseTimeZoneIntoMilliseconds = (timeZone: string): number => {
+    const match = timeZone.match(new RegExp('(?<sign>[\\+-])(?<hours>\\d{2}):(?<minutes>\\d{2})'));
+    const signStr = match?.groups?.['sign'];
+    const hoursStr = match?.groups?.['hours'];
+    const minutesStr = match?.groups?.['minutes'];
+
+    if (!signStr || !hoursStr || !minutesStr) {
+        return 0;
+    }
+
+    const base = 10;
+    const sign = signStr === '+' ? 1 : -1;
```

Comment:
Looks strange. You can parse hours directly with sign

Reply:
We need to apply sign to the sum of hours and minutes `sign * (millisInHour * hours + millisInMinute * minutes)`. I can't see how can I return the same result, while parsing sign together with hours